### PR TITLE
bpo-5846: Deprecate obsolete methods in `unittest`

### PR DESCRIPTION
**Pull request by erlend-aasland**
* `unittest.findTestCases`
  * `unittest.makeSuite`
  * `unittest.getTestCaseNames`

Scheduled for removal in Python 3.13

https://bugs.python.org/issue5846

### CODE REVIEW

**Comment by erlend-aasland** [comment]
Replaces GH-20400.

**Comment by serhiy-storchaka** [comment]
I am working on a patch which gets rid of uses of obsolete functions.

**Comment by erlend-aasland** [comment]
I had to fix `test_support`, since `__module__` is now `unittest` instead of `unittest.loader` for these three functions.

File: Doc/whatsnew/3.11.rst [hunk 1]
```diff
@@ -317,6 +317,21 @@ Deprecated
   It is untested and undocumented and also not used by webbrowser itself.
   (Contributed by Dong-hee Na in :issue:`42255`.)
 
+* Deprecated the following :mod:`unittest` functions, scheduled for removal in
+  Python 3.13:
+
+  * :meth:`unittest.findTestCases`
```
Comment (serhiy-storchaka) [inline Doc/whatsnew/3.11.rst]: :func:

File: Lib/test/test_support.py [hunk 2]
```diff
@@ -429,6 +429,7 @@ def test_check__all__(self):
         extra = {'TextTestResult', 'installHandler'}
         not_exported = {'load_tests', "TestProgram", "BaseTestSuite"}
 
+        extra = {'makeSuite', 'findTestCases', 'getTestCaseNames'}
```
Comment (serhiy-storchaka) [inline Lib/test/test_support.py]: `extra` is defined twice.

File: Lib/unittest/__init__.py [hunk 3]
```diff
@@ -70,6 +70,37 @@ def testMultiply(self):
 # deprecated
 _TextTestResult = TextTestResult
 
+from .loader import (
+    makeSuite as _makeSuite,
```
Comment (serhiy-storchaka) [inline Lib/unittest/__init__.py:74]: Why not add deprecations when they are defined?

**Review by serhiy-storchaka — COMMENTED** [review]
Are all tests warning-free?

If these functions are not used in unittest tests, add special tests, catch DeprecationWarning and check the `filename` attribute (to ensure that the stacklevel is correct).

File: Lib/unittest/__init__.py (hunk 3 above)
Comment (erlend-aasland) [reply Lib/unittest/__init__.py:74]: IIUC, it is the "shortcuts" that are deprecated, not the methods themselves. Should I deprecate the undocumented `unittest.TestLoader.makeSuite` and `unittest.TestLoader.findTestCases` as well?

Comment (serhiy-storchaka) [reply Lib/unittest/__init__.py:74]: There are no `unittest.TestLoader.makeSuite` and `unittest.TestLoader.findTestCases`.

Comment (erlend-aasland) [reply Lib/unittest/__init__.py:74]: Sorry, I meant `unittest.loader`, not `unittest.TestLoader`:

```
>>> import unittest
>>> unittest.loader.makeSuite
<function makeSuite at 0x10be0e2a0>
>>> unittest.loader.findTestCases
<function findTestCases at 0x10be0e340>
>>> 
```

**Comment by erlend-aasland** [comment]
Thanks for reviewing and merging, @serhiy-storchaka and @ambv!

**Pull request merged**